app/code/core/Mage/ImportExport/Model/Export/Entity/Product.php

```diff
@@ -926,7 +926,9 @@ protected function _prepareExport()
                         $dataRow[self::COL_TYPE]     = null;
                     } else {
                         $dataRow[self::COL_STORE] = null;
-                        $dataRow += $stockItemRows[$productId];
+                        if ($stockItemRows[$productId]) {
```

Comment:
Agree, export should continue, garbage in garbage out. Is this better:

```suggestion
                        if (!empty($stockItemRows[$productId])) {
```